applications/StructuralMechanicsApplication/custom_elements/small_displacement_bbar.cpp

```diff
@@ -0,0 +1,947 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Manuel Caicedo
+//                   Marcelo Raschi
+//                   Javier Mroginski
+
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "custom_elements/small_displacement_bbar.h"
+
+namespace Kratos
+{
+SmallDisplacementBbar::SmallDisplacementBbar(IndexType NewId,
+                                             GeometryType::Pointer pGeometry)
+        : BaseSolidElement(NewId, pGeometry)
+{
+    // DO NOT ADD DOFS HERE!!!
+}
+
+//************************************************************************************
+//************************************************************************************
+
+SmallDisplacementBbar::SmallDisplacementBbar(IndexType NewId,
+                                             GeometryType::Pointer pGeometry,
+                                             PropertiesType::Pointer pProperties)
+        : BaseSolidElement(NewId, pGeometry, pProperties)
+{
+}
+
+//************************************************************************************
+//************************************************************************************
+
+Element::Pointer SmallDisplacementBbar::Create(IndexType NewId,
```

Comment:
i would say you need to have two create methods. one with the geometry pointer and one with an array of nodes. take care, this may be quite important (take it as a TODO, i think we should accept for now)